.github/workflows/semgrep.yml

```diff
@@ -0,0 +1,24 @@
+name: Semgrep
+
+on:
+  # Scan changed files in PRs (diff-aware scanning):
+  pull_request: {}
+  # Scan on-demand through GitHub Actions interface:
+  workflow_dispatch: {}
+  # Scan mainline branches and report all findings:
+  push:
+    branches: ["master"]
+
+jobs:
+  semgrep:
+    name: semgrep/ci
+    runs-on: ubuntu-latest
+
+    container:
+      image: returntocorp/semgrep
+
+    steps:
+      - uses: actions/checkout@v3
+      - run: semgrep ci --include=bin/* --include=ext/* --include=lib/*
```

Comment:
Should we include the `lin-injectioin` folder as well? We do a bunch of shell commands that could lead to potential security attacks as well